### PR TITLE
Add semaphore to block on puppeteer chromium execution

### PR DESCRIPTION
**Pull request by joshuali925**
*Issue #, if available:*

*Description of changes:*
- currently number of resources for the semaphore is 1
- semaphore times out and front end fails after 180 seconds, if timeout UI will display this toast:
image
- for on demand it displays this toast depending on report source:
image

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: kibana-reports/server/routes/lib/createReport.ts [hunk 1]
```diff
@@ -102,13 +105,18 @@ export const createReport = async (
...
         });
       }
-      createReportResult = await createVisualReport(
-        reportParams,
-        completeQueryUrl,
-        logger,
-        cookieObject,
-        timezone
-      );
+      const [value, release] = await semaphore.acquire();
```
Comment (penghuo) [inline kibana-reports/server/routes/lib/createReport.ts:108]: what if the acquire failed/timeout? is there retry avaliable?

**Comment by penghuo** [comment]
Do we have the UT/IT to cover the change?

File: kibana-reports/server/plugin.ts [hunk 2]
```diff
@@ -50,9 +51,11 @@ export class OpendistroKibanaReportsPlugin
       OpendistroKibanaReportsPluginStart
     > {
   private readonly logger: Logger;
+  private readonly semaphore: SemaphoreInterface;
 
   constructor(initializerContext: PluginInitializerContext) {
     this.logger = initializerContext.logger.get();
+    this.semaphore = withTimeout(new Semaphore(1), 30000, new Error('timeout'));
```
Comment (akbhatta) [inline kibana-reports/server/plugin.ts]: I think 30 sec is too less a timeout. may be 180 seconds?

File: kibana-reports/server/routes/lib/createReport.ts (hunk 1 above)
Comment (akbhatta) [reply kibana-reports/server/routes/lib/createReport.ts:108]: I think retrying semaphore not required. we can just increase timeout to required value.

Comment (joshuali925) [reply kibana-reports/server/routes/lib/createReport.ts:108]: If request fails or timeout, client side would get an error. I think increasing timeout is better than client auto retry. I can improve the error message to indicate it's a timeout on UI

File: kibana-reports/server/plugin.ts (hunk 2 above)
Comment (joshuali925) [reply kibana-reports/server/plugin.ts]: changed to 180 seconds

**Review by akbhatta — APPROVED** [review]

**Review by zhongnansu — APPROVED** [review]

File: kibana-reports/public/components/main/main_utils.tsx [hunk 3]
```diff
@@ -198,6 +198,11 @@ export const generateReportById = async (
       console.log('error on generating report by id:', error);
       if (error.body.statusCode === 403) {
         handlePermissionsMissingToast();
+      } else if (error.body.statusCode === 503) {
+        handleErrorToast(
+          'Error generating report.',
+          `Timed out generating report ID ${reportId}. Try again later.`
```
Comment (penghuo) [inline kibana-reports/public/components/main/main_utils.tsx:204]: does 503 always because of timeout?

**Pull request merged**